invokeai/app/invocations/prompt_tools.py

```diff
@@ -0,0 +1,286 @@
+    )
+    append: bool = InputField(default=True, description="Append or overwrite file")
+
+    def invoke(self, context: InvocationContext) -> PromptsToFileInvocationOutput:
+        with open(self.file_path, "a" if self.append else "w") as f:
+            if isinstance(self.prompts, list):
+                for line in self.prompts:
+                    f.write(line + "\n")
+            else:
+                f.write((self.prompts or "") + "\n")
+
+        return PromptsToFileInvocationOutput()
+
+
+@invocation_output("prompt_pos_neg_output")
+class PromptPosNegOutput(BaseInvocationOutput):
+    """Base class for invocations that output a posirtive and negative prompt"""
+
+    positive_prompt: str = OutputField(description="Positive prompt")
+    negative_prompt: str = OutputField(description="Negative prompt")
+
+
+@invocation(
+    "prompt_split_neg",
+    title="Prompt Split Negative",
+    tags=["prompt", "split", "negative"],
+    category="prompt",
+    version="1.0.0",
+)
+class PromptSplitNegInvocation(BaseInvocation):
+    """Splits prompt into two prompts, inside [] goes into negative prompt everthing else goes into positive prompt. Each [ and ] character is replaced with a space"""
+
+    prompt: str = InputField(default="", description="Prompt to split")
+
+    def invoke(self, context: InvocationContext) -> PromptPosNegOutput:
+        p_prompt = ""
+        n_prompt = ""
+        brackets_depth = 0
+        escaped = False
+
+        for char in self.prompt or "":
+            if char == "[" and not escaped:
+                n_prompt += " "
+                brackets_depth += 1
+            elif char == "]" and not escaped:
+                brackets_depth -= 1
+                char = " "
+            elif brackets_depth > 0:
+                n_prompt += char
+            else:
+                p_prompt += char
+
+            # keep track of the escape char but only if it isn't escaped already
+            if char == "\\" and not escaped:
+                escaped = True
+            else:
+                escaped = False
+
+        return PromptPosNegOutput(positive_prompt=p_prompt, negative_prompt=n_prompt)
+
+
+@invocation("prompt_join", title="Prompt Join", tags=["prompt", "join"], category="prompt", version="1.0.0")
+class PromptJoinInvocation(BaseInvocation):
+    """Joins prompt left to prompt right"""
+
+    prompt_left: str = InputField(default="", description="Prompt Left", ui_component=UIComponent.Textarea)
+    prompt_right: str = InputField(default="", description="Prompt Right", ui_component=UIComponent.Textarea)
+
+    def invoke(self, context: InvocationContext) -> StringOutput:
+        return StringOutput(value=((self.prompt_left or "") + (self.prompt_right or "")))
+
+
+@invocation("prompt_join_three", title="Prompt Join Three", tags=["prompt", "join"], category="prompt", version="1.0.0")
+class PromptJoinThreeInvocation(BaseInvocation):
```

Comment:
I'd like to make this more generalized... it would be nice if there was just a single extensible string concat node that let you concat any number of strings, but that will need to come in a future change. For now, this is fine

Reply:
I totally agree an extensible version would be the ideal and would be very useful in many other nodes.  Additionally this functionality can be recreated using the existing single join version but I found that joining 3 things happened enough times that a separate node just made sense.    I assume this should also go into strings.py